### ISSUE TITLE
Getting DNLA to work in Docker

### ISSUE THREAD

**Issue by beardedfool** [airsonic-advanced/airsonic-advanced#681]
Bit lost trying to get DNLA working on the docker:latest using ports. It's not detected from other devices

It does seem to work if you set
`network_mode: "host"`

I've tried setting it with 
```
    environment:
      - UPnpPort=4041
```

Am I misunderstaning this, should it go in a a JAVA_OPTS?

Can I confirm that the LAN Server address in DNLA settings should be the web 
e.g. http://music:4040/
rather than the UPnp http://music:4041/

nmap is showing the ports as open on the host (closed when docker not running)

Interestingly changin the UPnpPort doesn't seem that change it in the docker

```
version: "3"

services:
  airsonic-advanced:
    container_name: airsonic-advanced
    image: airsonicadvanced/airsonic-advanced:latest
    restart: unless-stopped
    ports: 
      - 4040:4040
      - 4044:4044 #UPnp
      - 1900:1900 # DNLA
    user: 1207:1207
    environment:
      - TZ=Europe/London #optional - show logs in local timezone
      # https://airsonic.github.io/docs/configure/airsonic-properties/
      - UPnpPort=4044
      # - JAVA_OPTS=-Dserver.forward-headers-strategy=native #optional - if you use a reverse-proxy
    volumes:
      - /home/docker/airsonic/playlists:/var/playlists
      - /home/docker/airsonic/podcasts:/var/podcasts
      - /home/docker/airsonic/data:/var/airsonic
```

Gives 

```
airsonic-advanced    | 2021-12-02 13:16:34.736  INFO --- o.a.player.service.UPnPService           : Starting UPnP service...
airsonic-advanced    | 2021-12-02 13:16:34.753  INFO --- o.a.player.service.UPnPService           : Successfully started UPnP service on port 4041!
airsonic-advanced    | 2021-12-02 13:16:34.791  INFO --- o.a.player.service.UPnPService           : Enabling UPnP/DLNA media server
```

Is this relevant?
https://github.com/airsonic/airsonic/pull/1380

**Comment by beardedfool** [comment]
Just to add to this, I realised that some of the ports were udp so the docker-compose now reads
Appreciate I shouldn't need all of these
5000 was a wild stab in the dark

```
    ports: 
      - 4040:4040
      - 4041:4041/tcp #UPnp/tcp
      - 4041:4041/udp #UPnp/udp
      - 1900:1900/udp # DNLA/udp
      - 1900:1900/tcp # DNLA/tcp
      - 5000:5000/tcp
      - 5000:5000/udp
    environment:
      - UPnpPort=4041
```
      
      
but still can't see it

**Comment by randomnicode** [comment]
For docker, because the run script automatically binds the env variable, the actual env variable is: `UPNP_PORT`

That being said, the default is 4041 and 1900 already, so as long as you bind these externally, you shouldn't have to specify them in the environment.

**Comment by beardedfool** [comment]
Just to confirm that UPNP_PORT does indeed change the port, thanks

Can you just confirm whether that 'LAN Server address' in DNLA settings should be the http or Upnp port please?

Bit lost on this one. May give the Macvlan driver a go and see what happens there

**Comment by randomnicode** [comment]
Don't *quite* know the answer. Maybe you could figure out and report it here so we can have the answer definitively once and for all?

**Comment by mikeder** [comment]
I've been trying to get UPnP/DLNA to work inside Docker and behind a Traefik reverse proxy for a while now and the effort has lead me to 2 relevant Docker issues:

https://github.com/moby/moby/issues/23659
https://github.com/moby/libnetwork/issues/2397

TLDR; External multicast packets ( what is used to discover UPnP/DLNA on your network ) are not routed to docker containers properly unless you bind the container to the host network bridge. @beardedfool in your case this means add `network_mode: host` to your `docker-compose.yml`

I'm going to do the same and report back if I can get this working and still have Traefik handle reverse proxy duties for external access.
